packages/components/script.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { useRef, useEffect } from "react";
+import { connect } from "frontity";
+import { Connect, Package } from "frontity/types";
+
+type ScriptType = React.FC<Connect<Package, { src: string }>>;
+
+const generateId = () => Math.trunc(Math.random() * 9999999).toString();
+
+const Script: ScriptType = ({ state, ...props }) => {
+  const scriptId = useRef(generateId());
```

Comment:
Instead of `generateId` which will break hydration I would try to keep the reference to the `script`. Something like this:

```js
const Script: ScriptType = ({ state, ...props }) => {
  let script = null;
  const { src } = props;

  useEffect(() => {
    if (state.frontity.rendering === "csr") {
      script = window.document.createElement("script");
      script.src = src;
      script.async = true;
      script.charset = "utf-8";

      window.document.body.appendChild(script);
    }

    return () => {
      if (script) window.document.body.removeChild(script);
    };
  }, []);

  return <script async src={src} charSet="utf-8" />;
};
```